profiles/common/modules/features/multisite_notifications/multisite_notifications_core/multisite_notifications_core.module

```diff
@@ -75,74 +74,17 @@ function multisite_notifications_core_menu() {
 }
 
 /**
- * Implements hook_block_info().
- */
-function multisite_notifications_core_block_info() {
-  return array(
-    'notification_block' => array(
-      'info' => t('Anonymous subscription Block'),
-      'cache' => DRUPAL_CACHE_PER_PAGE,
-    ),
-  );
-}
-
-/**
- * Implements hook_block_view().
+ * Implements hook_form_alter().
  */
-function multisite_notifications_core_block_view($delta = '') {
-  global $user;
-  $block = array();
-  switch ($delta) {
-    case 'notification_block':
-      if ($user->uid == '0') {
-        $content = drupal_get_form('multisite_notifications_core_form');
-        $block['subject'] = t('Subscribe to new content');
-        $block['content'] = drupal_render($content);
-      }
-      break;
-
-    default:
-      break;
-  }
-  return $block;
-}
-
-/**
- * Call back function that returns the form to (un)subscribe as anonymous user.
- */
-function multisite_notifications_core_form() {
-
-  $form = array();
-  $form['mail'] = array(
-    '#type' => 'textfield',
-    '#size' => 30,
-    '#maxlength' => 128,
-    '#default_value' => 'Fill in your email address',
-    '#required' => TRUE,
-  );
-
-  $form['subscribe'] = array(
-    '#type' => 'submit',
-    '#value' => t('Subscribe'),
-    '#weight' => 20,
-  );
-  $form['unsubscribe'] = array(
-    '#type' => 'submit',
-    '#value' => t('Unsubscribe'),
-    '#weight' => 30,
-  );
+function multisite_notifications_core_form_workbench_moderation_moderate_form_alter(&$form, &$form_state) {
+  $form['#submit'][] = '_multisite_notifications_core_wf_submit';
```

Comment:
Instead of hooking an additional submit function into the form, we probably better implement hook_workbench_moderation_transition() which is called from workbench_moderation_moderate() and thus catches more cases than the moderation form.